File: Programmers/Lv1/Lv1_예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> d, int budget) {
	int answer = 0;
	sort(d.begin(), d.end());
	for (int i = 0; i < d.size(); i++) {
		budget -= d[i];
		if (budget >= 0)
			answer++;
		else
			break;
	}
	return answer;
}

File: Programmers/Lv1/Lv1_예산.py
def solution(d, budget):
    answer = 0
    d.sort()
    ans = 0
    for a in d:
        ans += a
        answer += 1
        if(ans>budget):
            answer -= 1
            break;
    return answer

print(solution([1,3,2,5,4],9))